src/result_processor.c

```diff
@@ -375,12 +375,20 @@ static int cmpByFields(const void *e1, const void *e2, const void *udata) {
   for (size_t i = 0; i < self->fieldcmp.nkeys && i < SORTASCMAP_MAXFIELDS; i++) {
     const RSValue *v1 = RLookup_GetItem(self->fieldcmp.keys[i], &h1->rowdata);
     const RSValue *v2 = RLookup_GetItem(self->fieldcmp.keys[i], &h2->rowdata);
+    // take the ascending bit for this property from the ascending bitmap
+    ascending = SORTASCMAP_GETASC(self->fieldcmp.ascendMap, i);
     if (!v1 || !v2) {
-      break;
+      int rc;
+      if(v1){
```

Comment:
1) I think a variable name like `cmpval` might be better here, since its ultimately returned on ascending
2) formatting

Reply:
@mnunberg I just continue with the same name as define in the entire function

Reply:
ok